lib/application/pages/home/collections/update/update_collection_page.dart

```diff
@@ -0,0 +1,166 @@
+
+  final _Segment selectedSegment;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final vm = useUpdateCollectionVM(ref);
+    final state = useUpdateCollectionState(ref);
+
+    if (state is UpdateCollectionFailedLoading) {
+      return Center(child: ExceptionRetryContainer(exception: state.exception, onRetry: vm.loadInitialContent));
+    }
+
+    switch (selectedSegment) {
+      case _Segment.details:
+        return UpdateCollectionDetails();
+      case _Segment.memos:
+        return UpdateCollectionMemos();
+    }
+  }
+}
+
+extension on _Segment {
+  String get title {
+    switch (this) {
+      case _Segment.details:
+        return strings.details;
+      case _Segment.memos:
+        return strings.memos;
+    }
+  }
+}
+
+class _BottomActionContainer extends ConsumerWidget {
+  const _BottomActionContainer({required this.selectedSegment, required this.onSegmentSwapRequested});
+
+  final _Segment selectedSegment;
+  final void Function(_Segment segment) onSegmentSwapRequested;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+
+    late Widget button;
+
+    switch (selectedSegment) {
+      case _Segment.details:
+        button = _DetailsActionButton(onSegmentSwapRequested: onSegmentSwapRequested);
+        break;
+      case _Segment.memos:
+        button = _MemosActionButton();
+        break;
+    }
+
+    return ThemedBottomContainer(
+      child: Container(
+        color: theme.neutralSwatch.shade800,
+        child: SafeArea(
+          child: button.withSymmetricalPadding(context, vertical: Spacing.small, horizontal: Spacing.medium),
+        ),
+      ),
+    );
+  }
+}
+
+class _DetailsActionButton extends ConsumerWidget {
+  const _DetailsActionButton({required this.onSegmentSwapRequested});
+
+  final Function(_Segment segment) onSegmentSwapRequested;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final state = useUpdateCollectionState(ref);
+
+    if (state is! UpdateCollectionLoaded) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    final vm = useUpdateCollectionVM(ref);
+
+    void onPressed() {
+      if (state.hasMemos) {
+        vm.saveCollection();
+      } else {
+        onSegmentSwapRequested(_Segment.memos);
+      }
+    }
+
+    final buttonTitle = state.hasMemos ? strings.saveCollection : strings.next;
+
+    return PrimaryElevatedButton(onPressed: state.hasDetails ? onPressed : null, text: buttonTitle.toUpperCase());
+  }
+}
+
+class _MemosActionButton extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final state = useUpdateCollectionState(ref);
+
+    if (state is! UpdateCollectionLoaded) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    final vm = useUpdateCollectionVM(ref);
+    final canSave = state.hasDetails && state.hasMemos;
```

Comment:
would expose this "canSave" directly from the vm, instead of assuming it from the widget (UI)

Reply:
You mean directly from the state, right? Created a getter there